docs/tutorials/gluon/performance.md

```diff
@@ -0,0 +1,485 @@
+
+Overall we see a substantial increase in training throughput: ~85% higher than full-precision training.
+
+## Tips & Tricks #5: Others
+
+Many other tips and tricks exist for optimizing the throughput of training.
+
+One area we didn't explicitly benchmark in this tutorial is data transfer from CPU to GPU memory. Usually this isn't an issue, but for very large arrays this can become a bottleneck too. You might be able to compress your data significantly before transferring if your data is sparse (i.e. mostly zero values). Check out the sparse array tutorial for more details and an example of how this can impact training speed.
+
+Another useful trick if data pre-processing or data transfer is the bottleneck is pre-fetching batches. You can write your training loop to transfer the next batch of data to GPU before processing the current batch. Once again, this trick is memory permitting.
+
+And finally, if you are an advanced user, check out the various [environment variables](https://mxnet.incubator.apache.org/faq/env_var.html) that can be configured to change the behaviour of the MXNet backend.
+
+## Final Benchmark
+
+We will now combine all of the above tricks and tips in the complete training loop and compare to the initial benchmark.
+
+
+```python
+iters = 25
+num_samples = 0
+num_iters = 0
+start_time = time.time()
+for iter_idx, (data, label) in enumerate(dataloader):
+    num_samples += data.shape[0]
+    num_iters += 1
+    data = data.as_in_context(ctx).astype('float16')
+    label = label.as_in_context(ctx).astype('float16')
+    with mx.autograd.record():
+        pred = net(data)
+        loss = loss_fn(pred, label)
+    loss.backward()
+    trainer.step(data.shape[0])
+    metric.update(label, pred)
+    print('.', end='')
+    if num_iters >= iters:
+        break
+end_time = time.time()
+total_time = end_time - start_time
+print('\n')
+print('average iterations/sec: {:.4f}'.format(num_iters/total_time))
+print('average samples/sec: {:.4f}'.format(num_samples/total_time))
+```
+
+    .........................
+    
+    average iterations/sec: 6.5968
+    average samples/sec: 422.1925
+
+
+Using the above tips and tricks we managed to increase the throughput of training by ~500% from the initial benchmark! Our training throughput is less than the throughput of the individual components we tested, but there are additional overheads that we didn't previously measure (such as data transfer to GPU).
+
+## Conclusion
+
+We learnt a number of tips and tricks to optimize the throughput of training, and they lead to a considerable increase compared to our initial baseline. As general rules, set `num_workers` on the `DataLoader` to >0, and hybridize your network if you're not debugging. You should increase `batch_size` where possible, but do this with care because of its potential effects on convergence. And finally, consider mixed precision training for substantial speed-ups if you're willing to accept a small drop in network accuracy.
```

Comment:
```suggestion
We learned a number of tips and tricks to optimize the throughput of training, and they lead to a considerable increase compared to our initial baseline. As general rules, set `num_workers` on the `DataLoader` to >0, and hybridize your network if you're not debugging. You should increase `batch_size` where possible, but do this with care because of its potential effects on convergence. And finally, consider mixed precision training for substantial speed-ups if you're willing to accept a small drop in network accuracy.
```

Reply:
Changed.